Update template vars doc (#2379)
influxdata/docs-v2: content/chronograf/v1.7/guides/dashboard-template-variables.md

    parent: Guides
---

Chronograf's dashboard template variables let you update cell queries without editing queries, making it easy to interact with your dashboard cells and explore your data.

## Using template variables

> Be careful when quoting template variables in your queries. When creating a custom meta query, template values are not automatically quoted. Quote template variables as follows:

Reviewer: I think i might try to describe it like this (and this might go into a separate section about quoting):

For **predefined meta queries** such as "Field Keys" and "Tag Values", Chronograf will determine the appropriate quoting for you, so you do NOT need to add quotes (single or double) to your queries:

```
example goes here
```

For **custom queries, CSV, or map queries** we leave it up to the user to decide when to quote the values in the query in accordance with standard InfluxQL syntax. For numerical values, you do not need to quote them anywhere.

For string values, you can choose to quote the values in the variable definition. For example, if you define a custom CSV variable named `host`, and define it with the following values:

```
'host1','host2','host3'
```

Then you would use them as is in the query the same way you would use the predefined queries, without quoting:

```sql
SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
  WHERE "host" = :host: and time > :dashboardTime
```

If you prefer, you can leave the values in the template variable as strings without quotes and quote them when used in the query. In this case, the `host` variable would be defined without quotes:

```
host1,host2,host3
```

and used in the query like this using single quotes for the value:

```sql
SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
  WHERE "host" = ':host:' and time > :dashboardTime
```

When using custom meta queries that return strings, you will almost always need to quote the values when using them in a dashboard query, as the results from InfluxQL come back without quotes.

Being able to control the quoting in the query can be useful if you are using template variable strings in regular expression syntax, where quotes would cause query syntax errors.

Author: thank you, @russorat!

- [Using template variables](#using-template-variables)
- [Quoting template variables](#quoting-template-variables)
- [Predefined template variables](#predefined-template-variables)
- [Create custom template variables](#create-custom-template-variables)
- [Template variable types](#template-variable-types)
- [Reserved variable names](#reserved-variable-names)
- [Advanced template variable usage](#advanced-template-variable-usage)

## Using template variables

Use either [predefined template variables](#predefined-template-variables)
or [custom template variables](#create-custom-template-variables).
Variable values are then selected in your dashboard user-interface (UI).

![Using template variables](/img/chronograf/1-6-template-vars-use.gif)

## Quoting template variables

For **predefined meta queries** such as "Field Keys" and "Tag Values", **do not add quotes** (single or double) to your queries. Chronograf adds quoting as follows:

```sql
  SELECT :variable_name: FROM "telegraf"."autogen".:measurement: WHERE time < :dashboardTime:

> ``````

Reviewer 2: ~~~suggestion
```sql
SELECT :variable_name: FROM "telegraf"."autogen".:measurement: WHERE time < :dashboardTime:
```
~~~

> 
> For **custom queries**, **CSV**, or **map queries**, you must quote the values in the query in accordance with standard [InfluxQL](/influxdb/v1.8/query_language/) syntax as follows:
> .
> - For numerical values, **do not quote**.
> - For string values, you can choose to quote the values in the variable definition (or not). For example, if you define a custom CSV variable named `host` with the following values:
> 

Reviewer 2: ```suggestion

For **custom queries**, **CSV**, or **map queries**, quote the values in the query in accordance with standard [InfluxQL](/influxdb/v1.8/query_language/) syntax as follows:

- For numerical values, **do not quote**.
- For string values, choose to quote the values in the variable definition (or not).

For example, if you define a custom CSV variable named `host` with the following values:

```

```sh
'host1','host2','host3'
```

You do not include quotes in the query:

```sql
  SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
  WHERE "host" = :host: and time > :dashboardTime

> ```

Reviewer 2: ~~~suggestion
```sql
SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
WHERE "host" = :host: and time > :dashboardTime
```
~~~

> Alternatively, leave the template variable values as strings without quotes, and then quote the variables in your queries. In this case, you define the `host` variable would be defined without quotes:

Reviewer 2: ```suggestion
Alternatively, leave the template variable values as strings without quotes, and then quote the variables in your queries. In this case, you define the `host` variable using unquoted values:
```

```sh
host1,host2,host3
```

And then add single quotes in the query as follows:

```sql
SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
  WHERE "host" = ':host:' and time > :dashboardTime

> ```

Reviewer 2: ~~~suggestion
```sql
SELECT mean("usage_user") AS "mean_usage_user" FROM "telegraf"."autogen"."cpu" 
WHERE "host" = ':host:' and time > :dashboardTime
```
~~~
